File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/glue/src/main/java/com/example/glue/GetJobRun.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.glue;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.glue.GlueClient;
import software.amazon.awssdk.services.glue.model.GetJobRunRequest;
import software.amazon.awssdk.services.glue.model.GetJobRunResponse;
import software.amazon.awssdk.services.glue.model.GlueException;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class GetJobRun {

    public static void main(String[] args) {

        final String USAGE = "\n" +
                "Usage:\n" +
                "    <jobName> <runId>\n\n" +
                "Where:\n" +
                "    jobName - the name of the job. \n" +
                "    runId - the run id value. \n";

        if (args.length != 2) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String jobName = args[0];
        String runId = args[1];

        Region region = Region.US_EAST_1;
        GlueClient glueClient = GlueClient.builder()
                .region(region)
                .build();

        getGlueJobRun(glueClient, jobName, runId);
        glueClient.close();
    }

    public static void getGlueJobRun(GlueClient glueClient, String jobName, String runId) {

        try {
              GetJobRunRequest jobRunRequest = GetJobRunRequest.builder()
                .jobName(jobName)
                .runId(runId)
                .build();

              GetJobRunResponse runResponse = glueClient.getJobRun(jobRunRequest);
              System.out.println("Job status is : "+runResponse.jobRun().jobRunStateAsString());
        } catch (GlueException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/glue/src/main/kotlin/com/kotlin/glue/GetJobRun.kt
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.glue

import aws.sdk.kotlin.services.glue.GlueClient
import aws.sdk.kotlin.services.glue.model.GetJobRunRequest
import kotlin.system.exitProcess

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args:Array<String>) {

    val usage = """
    Usage:
        <jobName> <runId>

    Where:
        jobName - the name of the job. 
        runId - the run id value that you can obtain from the AWS Management Console. 
    """

    if (args.size != 2) {
        println(usage)
        exitProcess(0)
    }

    val jobName = args[0]
    val runId = args[1]
    getGlueJobRun(jobName, runId)
    }

suspend fun getGlueJobRun(jobNameVal: String?, runIdVal: String?) {

    val request = GetJobRunRequest {
        jobName = jobNameVal
        runId = runIdVal
    }

    GlueClient { region = "us-east-1" }.use { glueClient ->
          val runResponse = glueClient.getJobRun(request)
          println("Job status is ${runResponse.jobRun?.toString()}")
    }
}
